src/python/tests/unittests/test_controller/test_scan/test_scanner_process.py

```
@@ -123,54 +117,103 @@ def _scan():
+    def test_spawned_process_propagates_fatal_error(self):
+        """Spawn a ScannerProcess with a fatal scanner and verify the exception propagates."""
+        scanner = FatalScanner()
+        process = ScannerProcess(scanner=scanner, interval_in_ms=0)
+
+        log_queue = multiprocessing.Queue()
+        process.set_mp_log_queue(log_queue, logging.DEBUG)
+
+        process.start()
+        process.join(timeout=5)
+
+        with self.assertRaises(ScannerError) as ctx:
```

Ruff (0.15.5)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)